### PR TITLE
[IZPACK-952] Avoid functional dependency on TargetPanel - allow installers without TargetPanel

### PR DESCRIPTION
**Pull request by hiteshbosamiya**
Fixes following issue as well:
[IZPACK-853] PacksPanel empty without activating TargetPanel before

[IZPACK-853]: https://izpack.atlassian.net/browse/IZPACK-853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by reinhapa** [comment]
@akuhtz I'm not that familiar with this piece of code, so I like that you also take a look at it.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java [hunk 1]
```diff
@@ -69,8 +73,15 @@ private void run(Runnable action)
             {
                 SwingUtilities.invokeAndWait(action);
             }
+            catch (InvocationTargetException e) {
+                logger.info("InvocationTargetException");
+                Throwable targetException = e.getTargetException();
+                targetException.printStackTrace();
```
Comment (akuhtz) [inline izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java:79]: Remove printStackTrace(). If stacktrace is required use Logger.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java [hunk 2]
```diff
@@ -69,8 +73,15 @@ private void run(Runnable action)
...
+                throw new IzPackException(targetException);
+            }
             catch (Exception exception)
             {
+                exception.printStackTrace();
```
Comment (akuhtz) [inline izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java:84]: Remove printStackTrace(). If stacktrace is required use Logger.

**Review by akuhtz — CHANGES_REQUESTED** [review]

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java (hunk 1 above)
Comment (hiteshbosamiya) [reply izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java]: Done

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java (hunk 2 above)
Comment (hiteshbosamiya) [reply izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java]: Done

**Comment by hiteshbosamiya** [comment]
Done the requested changes.

**Review by akuhtz — APPROVED** [review]

File: izpack-gui/src/main/java/com/izforge/izpack/gui/IzPanelLayout.java [hunk 3]
```diff
@@ -106,7 +106,6 @@
         catch (ClassNotFoundException e)
         {
             pathSelectionPanel = null;
-            e.printStackTrace();
```
Comment (reinhapa) [inline izpack-gui/src/main/java/com/izforge/izpack/gui/IzPanelLayout.java:109]: For better analysis the catched exception still should be logged in maybe fine mode using the 
`java.util.logging.Logger#log(Level level, String msg, Throwable thrown)` method.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java [hunk 4]
```diff
@@ -65,12 +71,22 @@ private void run(Runnable action)
...
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
             try
             {
                 SwingUtilities.invokeAndWait(action);
             }
+            catch (InvocationTargetException e) {
+                Throwable targetException = e.getTargetException();
+                targetException.printStackTrace(pw);
+                logger.info("InvocationTargetException: " + sw);
```
Comment (reinhapa) [inline izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java:83]: Better use the following method for logging in order the logging framework to handle the stacktrace output:
`java.util.logging.Logger#log(Level level, String msg, Throwable thrown)` as so:

```
logger.log(Level.INFO, "Action invocation failed", targetException);
```

**Review by reinhapa — CHANGES_REQUESTED** [review]

File: izpack-gui/src/main/java/com/izforge/izpack/gui/IzPanelLayout.java (hunk 3 above)
Comment (hiteshbosamiya) [reply izpack-gui/src/main/java/com/izforge/izpack/gui/IzPanelLayout.java:109]: Done.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java (hunk 4 above)
Comment (hiteshbosamiya) [reply izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerController.java]: Done.

File: izpack-installer/src/main/java/com/izforge/izpack/installer/unpacker/UnpackerBase.java [hunk 5]
```diff
@@ -445,6 +447,16 @@ public synchronized boolean isInterruptDisabled()
     protected void preUnpack(List<Pack> packs) throws InstallerException
     {
         logger.fine("Unpacker starting");
+        String installPath = installData.getInstallPath();
+        if (installPath == null)
+        {
```
[the remark this answers is not in the document]
Comment (hiteshbosamiya) [reply izpack-installer/src/main/java/com/izforge/izpack/installer/unpacker/UnpackerBase.java:452]: We need this check here as `installData.setInstallPath(path);` gets called in `DefaultTargetPanel`, `TargetPanel` & other related classes based on the value returned by `installData.getInstallPath();`.

**Comment by hiteshbosamiya** [comment]
@akuhtz, any idea why the build fails for maven-plugin-plugin?

**Comment by hiteshbosamiya** [comment]
@reinhapa, please refer to my reply.

**Review by reinhapa — APPROVED** [review]

**Pull request merged**